src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -779,7 +779,7 @@ MatterThermostatClusterServerPreAttributeChangedCallback(const app::ConcreteAttr
         }
         auto RequestedSystemMode = static_cast<SystemModeEnum>(*value);
         if (ControlSequenceOfOperation > ControlSequenceOfOperationEnum::kCoolingAndHeatingWithReheat ||
-            RequestedSystemMode > SystemModeEnum::kFanOnly)
+            RequestedSystemMode > SystemModeEnum::kSleep)
```

Comment:
We're actually hoping to deprecate/remove these modes, as they're better handled elsewhere and were just a carryover from Zigbee.

Reply:
Reverted.